x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -47,25 +48,10 @@ const BASE_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET = zodMaskFor<BaseCreateProps>(
   'outcome',
 ]);
 
-/**
- * Aditionally remove fields which are part only of the optional fields in the rule types that make up
- * the TypeSpecificCreateProps discriminatedUnion, by recreating a discriminated union of the types, but
- * with the necessary fields omitted, in the types where they exist. Fields to extract:
- *  - response_actions: from Query and SavedQuery rules
- */
-const TYPE_SPECIFIC_FIELDS_TO_OMIT = ['response_actions'] as const;
-
-const TYPE_SPECIFIC_FIELDS_TO_OMIT_FROM_QUERY_RULES = zodMaskFor<QueryRuleCreateFields>()([
-  ...TYPE_SPECIFIC_FIELDS_TO_OMIT,
-]);
-const TYPE_SPECIFIC_FIELDS_TO_OMIT_FROM_SAVED_QUERY_RULES =
-  zodMaskFor<SavedQueryRuleCreateFields>()([...TYPE_SPECIFIC_FIELDS_TO_OMIT]);
-
-export type TypeSpecificFields = z.infer<typeof TypeSpecificFields>;
 export const TypeSpecificFields = z.discriminatedUnion('type', [
   EqlRuleCreateFields,
-  QueryRuleCreateFields.omit(TYPE_SPECIFIC_FIELDS_TO_OMIT_FROM_QUERY_RULES),
-  SavedQueryRuleCreateFields.omit(TYPE_SPECIFIC_FIELDS_TO_OMIT_FROM_SAVED_QUERY_RULES),
+  QueryRuleCreateFields,
```

Comment:
Since no fields are omitted from any of these rule types, `TypeSpecificFields` is now exactly the same as `TypeSpecificCreatePropsInternal` from `x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/rule_schemas.gen.ts` (you might need to rebase from main to be able to import this Zod schema).

So `TypeSpecificFields` doesn't need to exist anymore, and neither does the `areTypesEqual` test below.

So below, PrebuiltRuleAsset can just be:
```ts
export const PrebuiltRuleAsset = PrebuiltAssetBaseProps.and(TypeSpecificCreatePropsInternal).and(
  z.object({
    rule_id: RuleSignatureId,
    version: RuleVersion,
  })
);
```
Not sure if Zod will accept using `TypeSpecificCreateProps` directly instead of `TypeSpecificCreatePropsInternal`. Give it at try, if you see any issues, it's OK to use `TypeSpecificCreatePropsInternal`.

Reply:
Changed to `TypeSpecificCreatePropsInternal` - beacuse `TypeSpecificCreateProps` was not accepted as you mentioned. Given this change I removed the test that was comparing `TypeSpecificFields` and `TypeSpecificCreatePropsInternal`. Is that ok ?

Reply:
Yes, that's OK. Thanks!